sql/core/src/test/scala/org/apache/spark/sql/execution/SparkSqlParserSuite.scala

```diff
@@ -800,4 +800,39 @@ class SparkSqlParserSuite extends AnalysisTest with SharedSparkSession {
         start = 0,
         stop = 63))
   }
+
+  def getMemoryUsage: Long = {
+    System.gc()
+    Runtime.getRuntime.totalMemory() - Runtime.getRuntime.freeMemory()
+  }
+
+  def awfulQuery(depth: Int): String = {
+    if (depth == 0) {
+      s"rand()"
+    } else {
+      s"case when ${awfulQuery(depth - 1)} > 0.5 " +
+      s"then ${awfulQuery(depth - 1)} " +
+      s"else ${awfulQuery(depth - 1)} " +
+      "end"
+    }
+  }
+
+  test("release ANTLR cache after parsing") {
```

Comment:
If you don't mind, please use the test prefix like the following.
```scala
- test("release ANTLR cache after parsing") {
+ test("SPARK-47404: release ANTLR cache after parsing") {
```

Reply:
Done.